src/top_model/main.cpp

```diff
@@ -209,5 +215,22 @@ std::shared_ptr<cadmium::dynamic::modeling::coupled<TIME>> TOP = std::make_share
     r.run_until(NDTime("04:00:00:000"));
     auto elapsed = std::chrono::duration_cast<std::chrono::duration<double, std::ratio<1>>>(hclock::now() - start).count();
     cout << "Simulation took:" << elapsed << "sec" << endl;
+	
+	char input_file[] = "../data/abp_output.txt";
+	char output_file[] = "../data/abp_processed_output.csv";
+	/**
+	* the function taking input file &
+	* transfering into output file 
+	*/
+	
+	filter(input_file,output_file);
+
+	/**
+	* the function taking input file &
+	* transfering into output file 
+	*/
+	
+	filter2(input_file,output_file);
```

Comment:
You have to modify the input and output file name. These cannot be the same for both the functions. The output of` filter1 `becomes, the input of` filter2`. Also, create a new file name for second function `output_file`. Otherwise, it will be opening and writing the same file, which may result in an error.

Reply:
Done

Reply:
Yes It is done